DataAccessLayer/Repository/Implementations/CustomerRepository.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Common.Models;
+using DataAccessLayer.Exceptions;
+using DataAccessLayer.Repository.Interfaces;
+
+namespace DataAccessLayer.Repository.Implementations;
+
+public class CustomerRepository(TicketSpotDbContext ticketSpotContext)
+    : BaseRepository<Customer>(ticketSpotContext), ICustomerRepository
+{
+    public async Task<Customer> CreateAsync(Customer customer)
+    {
+        await _context.Customers.AddAsync(customer);
+        await _context.SaveChangesAsync();
+        return customer;
+    }
+
+    public async Task<Customer> UpdateAsync(Customer customer)
+    {
+        var itemToUpdate = await Get(x => x.Id == customer.Id)
+            ?? throw new RecordNotFoundException("The customer to be updated is not found in the database");
+
+        _context.Customers.Update(customer);
+        await _context.SaveChangesAsync();
+        return customer;
+    }
+
+    public async Task DeleteAsync(int id)
```

Comment:
You do not need to check if the item to delete exists according to the idempotency principle. https://blog.dreamfactory.com/what-is-idempotency